### PR TITLE
Feature/add mvvmlight

### PR DESCRIPTION
**Pull request by mariusbob**
Added MVVM-Light to and made it work throughout all the platforms.

### CODE REVIEW

File: SportsPlanner.Droid/MainActivity.cs [hunk 1]
```diff
@@ -1,44 +1,63 @@
...
 			base.OnCreate(bundle);
-			_eventsGateway = new EventsApiGateway();
 
 			// Set our view from the "main" layout resource
 			SetContentView(Resource.Layout.Main);
 
-			// Get our button from the layout resource,
-			// and attach an event to it
-			Button button = FindViewById<Button>(Resource.Id.myButton);
+            GetEventsBtn.SetCommand("Click", Vm.GetEventsCommand);
```
Comment (cosminstirbu) [inline SportsPlanner.Droid/MainActivity.cs]: Is there any other approach that doesn't require hardcoded "magic" strings?

File: SportsPlanner.Droid/MainActivity.cs [hunk 2]
```diff
@@ -1,44 +1,63 @@
...
+            get
+            {
+                return _pageTitle
+                    ?? (_pageTitle = FindViewById<TextView>(Resource.Id.pageTitle));
+            }
+        }
+
+        private Button _getEventsBtn;
+
+        public Button GetEventsBtn
```
Comment (cosminstirbu) [inline SportsPlanner.Droid/MainActivity.cs]: We can probably move all the "find by views" in a partial class like in the video you sent on Slack.

File: SportsPlanner.Droid/SportsPlanner.Droid.csproj.bak [hunk 3]
```diff
@@ -0,0 +1,74 @@
+﻿<?xml version="1.0" encoding="utf-8"?>
```
Comment (cosminstirbu) [inline SportsPlanner.Droid/SportsPlanner.Droid.csproj.bak]: Not sure this needs to be committed

File: SportsPlanner.Droid/SportsPlanner.Droid.csproj.user [hunk 4]
```diff
@@ -0,0 +1,8 @@
+﻿<?xml version="1.0" encoding="utf-8"?>
```
Comment (cosminstirbu) [inline SportsPlanner.Droid/SportsPlanner.Droid.csproj.user]: This seems like a file we would want to add to .gitignore

File: SportsPlanner.Droid/App.cs [hunk 5]
```diff
@@ -0,0 +1,27 @@
...
+
+        public static ViewModelLocator Locator
+        {
+            get
+            {
+                if (locator == null)
+                {
+                    // Initialize the MVVM Light DispatcherHelper.
+                    // This needs to be called on the UI thread.
+                    DispatcherHelper.Initialize();
```
Comment (cosminstirbu) [inline SportsPlanner.Droid/App.cs:18]: It looks like we call this just on Android? Any specific reason why this is different from iOS ?

File: SportsPlanner.iOS/SportsPlanner.iOS.csproj.user [hunk 6]
```diff
@@ -2,5 +2,6 @@
 <Project ToolsVersion="14.0" xmlns="http://schemas.microsoft.com/developer/msbuild/2003">
```
Comment (cosminstirbu) [inline SportsPlanner.iOS/SportsPlanner.iOS.csproj.user]: This seems like a file we would want to add to .gitignore

File: SportsPlanner.iOS/ViewController.cs [hunk 7]
```diff
@@ -1,48 +1,35 @@
...
-			var alertController = new UIAlertController();
-			alertController.Title = "Search Events Result";
-			alertController.Message = String.Format("Did receive the {0} events", searchEventsResult.Details.Total);
 
-			var dismissAction = UIAlertAction.Create("Dismiss", UIAlertActionStyle.Cancel, null);
-			alertController.AddAction(dismissAction);
-
-			PresentViewController(alertController, true, null);
-		}
+            Button.SetCommand("TouchUpInside", Vm.GetEventsCommand);
```
Comment (cosminstirbu) [inline SportsPlanner.iOS/ViewController.cs]: Is there a way we can avoid these magic strings?

File: SportsPlanner/ViewModels/MainViewModel.cs [hunk 8]
```diff
@@ -0,0 +1,50 @@
+﻿using GalaSoft.MvvmLight;
+using GalaSoft.MvvmLight.Command;
+using GalaSoft.MvvmLight.Views;
+using SportsPlanner.Gateways;
+
+namespace SportsPlanner.ViewModels
+{
+    public class MainViewModel : ViewModelBase
+    {
+        private IEventsGateway _iEventsGateway;
```
Comment (cosminstirbu) [inline SportsPlanner/ViewModels/MainViewModel.cs]: No need for the i in front of the variable name, same for the rest of the file.

File: SportsPlanner/ViewModels/ViewModelLocator.cs [hunk 9]
```diff
@@ -0,0 +1,47 @@
...
+            ////else
+            ////{
+            ////    // Create run time view services and models
+            ////    SimpleIoc.Default.Register<IDataService, DataService>();
+            ////}
+
+            SimpleIoc.Default.Register<MainViewModel>();
+
+            SimpleIoc.Default.Register<IEventsGateway, EventsApiGateway>();
+            SimpleIoc.Default.Register<IDialogService, DialogService>();
```
Comment (cosminstirbu) [inline SportsPlanner/ViewModels/ViewModelLocator.cs:31]: Does the DialogService as an iOS / Android and UWP implementation?
If so, that's pretty cool.

**Review by cosminstirbu — CHANGES_REQUESTED** [review]

File: SportsPlanner.Droid/App.cs (hunk 5 above)
Comment (mariusbob) [reply SportsPlanner.Droid/App.cs:18]: i'm not sure 100% how it works, but it definitely should be used in iOS project too.

File: SportsPlanner.Droid/MainActivity.cs (hunk 1 above)
Comment (mariusbob) [reply SportsPlanner.Droid/MainActivity.cs]: Maybe we can avoid the hardcoded strings by `nameOf`
https://msdn.microsoft.com/en-us/library/dn986596.aspx

File: SportsPlanner.iOS/SportsPlanner.iOS.csproj.user (hunk 6 above)
Comment (mariusbob) [inline SportsPlanner.iOS/SportsPlanner.iOS.csproj.user]: Maybe we can avoid the hardcoded strings by `nameOf`
https://msdn.microsoft.com/en-us/library/dn986596.aspx

File: SportsPlanner.iOS/ViewController.cs (hunk 7 above)
Comment (mariusbob) [reply SportsPlanner.iOS/ViewController.cs]: Maybe we can avoid the hardcoded strings by `nameOf`
https://msdn.microsoft.com/en-us/library/dn986596.aspx

File: SportsPlanner/ViewModels/MainViewModel.cs (hunk 8 above)
Comment (mariusbob) [reply SportsPlanner/ViewModels/MainViewModel.cs]: you're right

File: SportsPlanner/ViewModels/ViewModelLocator.cs (hunk 9 above)
Comment (mariusbob) [reply SportsPlanner/ViewModels/ViewModelLocator.cs:31]: Actually, MVVM Light has different implementation for both, iOS and Android platforms. You don't have to take care about the platform, mvvm light does it for you.

File: SportsPlanner.Droid/MainActivity.cs (hunk 2 above)
Comment (mariusbob) [reply SportsPlanner.Droid/MainActivity.cs]: indeed.

File: SportsPlanner.iOS/SportsPlanner.iOS.csproj.user (hunk 6 above)
Comment (mariusbob) [reply SportsPlanner.iOS/SportsPlanner.iOS.csproj.user]: done

File: SportsPlanner.Droid/SportsPlanner.Droid.csproj.bak (hunk 3 above)
Comment (mariusbob) [reply SportsPlanner.Droid/SportsPlanner.Droid.csproj.bak]: added it to gitignore

File: SportsPlanner.iOS/AppDelegate.cs [hunk 10]
```diff
@@ -17,9 +18,13 @@ public class AppDelegate : UIApplicationDelegate
...
 		public override bool FinishedLaunching (UIApplication application, NSDictionary launchOptions)
 		{
-			// Override point for customization after application launch.
-			// If not required for your application you can safely delete this method
-			return true;
+            // Override point for customization after application launch.
+            // If not required for your application you can safely delete this method
+
+            // MVVM Light's DispatcherHelper for cross-thread handling.
+            DispatcherHelper.Initialize(application);
```
Comment (CatalinPrata) [inline SportsPlanner.iOS/AppDelegate.cs:25]: Could you please format this method?

File: SportsPlanner.iOS/ViewController.cs [hunk 11]
```diff
@@ -1,48 +1,35 @@
...
+    public partial class ViewController : UIViewController
 	{
-		private IEventsGateway _eventsGateway;
-
-		public ViewController(IntPtr handle) : base(handle)
+        // Keep track of bindings to avoid premature garbage collection
+        private readonly List<Binding> bindings = new List<Binding>();
+
+        // Gets a reference to the MainViewModel from the ViewModelLocator.
+        private MainViewModel Vm
```
Comment (CatalinPrata) [inline SportsPlanner.iOS/ViewController.cs]: What is Vm? :P maybe a better naming would be better.

**Review by cosminstirbu — APPROVED** [review]

**Review by CatalinPrata — CHANGES_REQUESTED** [review]

**Review by CatalinPrata — APPROVED** [review]

**Pull request merged**